internal/controller/constraint_controller.go

```diff
@@ -0,0 +1,174 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	templatesv1 "github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/status/v1beta1"
+	constraintstatusv1beta1 "github.com/open-policy-agent/gatekeeper/v3/apis/status/v1beta1"
+	console "github.com/pluralsh/console-client-go"
+	consoleclient "github.com/pluralsh/deployment-operator/pkg/client"
+	"github.com/pluralsh/polly/algorithms"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const missingLabelError = "missing label while attempting to map a constraint status resource"
+
+type ConstraintReconciler struct {
+	client.Client
+	Scheme        *runtime.Scheme
+	ConsoleClient consoleclient.Client
+	Reader        client.Reader
+	Constrains    map[string]*console.PolicyConstraintAttributes
```

Comment:
I think this should be spelled `Constraints` everywhere

Reply:
done